cmake/onnxruntime_unittests.cmake

```diff
@@ -228,6 +233,10 @@ endif()
 
 file(GLOB onnxruntime_test_framework_src ${onnxruntime_test_framework_src_patterns})
 
+if(NOT (HAS_FILESYSTEM_H OR HAS_EXPERIMENTAL_FILESYSTEM_H))
```

Comment:
Let me submit a PR that totally remove header_files_test.cc. Then you don't need to change this cmake file.